frontend/test/pytest/test_jax_integration.py

```diff
@@ -0,0 +1,320 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Test QJIT compatibility with JAX transformations such as jax.jit and jax.grad."""
+
+from functools import partial
+
+import jax
+import jax.numpy as jnp
+import pennylane as qml
+import pytest
+
+from catalyst import measure, qjit
+
+
+class TestJAXJIT:
+    """Test QJIT compatibility with JAX compilation."""
+
+    def test_simple_circuit(self):
+        """Test a basic use case of jax.jit on top of qjit."""
+
+        @qjit
+        @qml.qnode(qml.device("lightning.qubit", wires=2))
```

Comment:
Could you also update these tests after merging with the main branch as follows?
```suggestion
    def test_simple_circuit(self, backend):
        """Test a basic use case of jax.jit on top of qjit."""

        @qjit
        @qml.qnode(qml.device(backend, wires=2))
```

Reply:
Thanks, will do!

Reply:
a521c48c2d27054bc58226fa5937f2bc8791ec08